### DISCUSSION TITLE
render --to pdf and ::: {.content-visible when-profile="MYPROFILE"}: introduces unwanted horizontal space in output

### DISCUSSION THREAD

**Discussion by r-cheologist** [quarto-dev/quarto-cli#4886] [Q&A]
Consider a `*.qml` file like this:

    Profile-dependent content follows:

    ::: {.content-visible when-profile="MYPROFILE"}
    Content requiring `--profile MYPROFILE` to be added to a `render` call
    ::::
    ::: {.content-visible when-profile="MYPROFILE"}
    More content requiring `--profile MYPROFILE` to be added to a `render` call
    ::::

This introduces extra hspace between the two containers, which prevents fine grained (non-paragraph spanning) use for example for profiles used for translation into different natural languages.

Can this be prevented?

**Comment by cscheid** [comment]
> which prevents fine grained (non-paragraph spanning) 

That's not currently a supported mode for `content-visible` and `content-hidden`. It's a good idea, though. We'll take a look at it for 1.4.

**Reply by cscheid** [reply]
Tracking here https://github.com/quarto-dev/quarto-cli/issues/4887

**Comment by mcanouil** [comment]
Is it on purpose the fenced div delimiters are not matched properly, _i.e., the fenced divs start with `:::` and end with `::::`?

**Reply by cscheid** [reply]
I think it's just a style choice (I agree that it reads strangely to me). But for what's worth, it's totally allowed by Pandoc:

```
$ cat divs.qmd
Hello

::: {.some-stuff}
content
::::
::: {.more-stuff}
more content
::::
$ quarto pandoc -f markdown -t native divs.qmd
[ Para [ Str "Hello" ]
, Div
    ( "" , [ "some-stuff" ] , [] ) [ Para [ Str "content" ] ]
, Div
    ( "" , [ "more-stuff" ] , [] )
    [ Para [ Str "more" , Space , Str "content" ] ]
]
```

I only recently learned that the marker of a div opening is the set of attributes next to the fence, not the matched number of colons.

**Reply by mcanouil** [reply]
Oh!

For nested fenced divs, this would be somewhat unreadable in my opinion.

**Comment by r-cheologist** [comment]
Aahhh. It really was an oversight svb making my example markdown...
